pkg/api/unversioned/group_version.go

```diff
@@ -64,12 +64,11 @@ func (gr *GroupResource) String() string {
 // ParseGroupResource turns "resource.group" string into a GroupResource struct.  Empty strings are allowed
 // for each field.
 func ParseGroupResource(gr string) GroupResource {
-	s := strings.SplitN(gr, ".", 2)
-	if len(s) == 1 {
-		return GroupResource{Resource: s[0]}
+	if i := strings.Index(gr, "."); i == -1 {
+		return GroupResource{Resource: gr}
+	} else {
+		return GroupResource{Group: gr[i+1:], Resource: gr[:i]}
```

Comment:
And now there are tests!

On Wed, Apr 27, 2016 at 1:29 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> And to soothe your mind, I'll add the tests for parsegroupversion that
> don't exist today
>
> On Wed, Apr 27, 2016 at 1:19 PM, David Eads <notifications@github.com>
> wrote:
>
>> In pkg/api/unversioned/group_version.go
>> <https://github.com/kubernetes/kubernetes/pull/24870#discussion_r61299159>
>> :
>>
>> > @@ -64,12 +64,11 @@ func (gr *GroupResource) String() string {
>> >  // ParseGroupResource turns "resource.group" string into a GroupResource struct.  Empty strings are allowed
>> >  // for each field.
>> >  func ParseGroupResource(gr string) GroupResource {
>> > -	s := strings.SplitN(gr, ".", 2)
>> > -	if len(s) == 1 {
>> > -		return GroupResource{Resource: s[0]}
>> > +	if i := strings.Index(gr, "."); i == -1 {
>> > +		return GroupResource{Resource: gr}
>> > +	} else {
>> > +		return GroupResource{Group: gr[i+1:], Resource: gr[:i]}
>>
>> Does this work properly when i==len-1? I'd expect an index out of bounds.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/24870/files/d5276ac4486c4e7529e84da91de6b12f0f3b667d#r61299159>
>>
>
>